python/pyspark/sql/plot/plotly.py

```diff
@@ -214,3 +209,35 @@ def plot_histogram(data: "DataFrame", **kwargs: Any) -> "Figure":
     fig["layout"]["xaxis"]["title"] = "value"
     fig["layout"]["yaxis"]["title"] = "count"
     return fig
+
+
+def process_column_param(column: Optional[Union[str, List[str]]], data: "DataFrame") -> List[str]:
+    """
+    Processes the provided column parameter for a DataFrame.
+    - If `column` is None, returns a list of numeric or datetime columns from the DataFrame.
+    - If `column` is a string, converts it to a list first.
+    - If `column` is a list, it checks if all specified columns exist in the DataFrame
+      and are of valid types (NumericType, DateType, or TimestampType).
+    - Raises a PySparkTypeError if any column in the list is not present in the DataFrame
+      or has an invalid type.
+    """
+    valid_types = (NumericType, DateType, TimestampType)
```

Comment:
Does these plot actually accept `DateType` and `TimestampType`?

it seems we only support `NumericType` in the PS side?
https://github.com/apache/spark/blob/54edfd3cdb1b2325b04734c3b9bdc289598de558/python/pyspark/pandas/plot/plotly.py#L156-L160

Reply:
Good catch!
Box plots seem to support NumericType only.
Density and histogram plots will accept `np.datetime64` in PS, as https://github.com/apache/spark/blob/master/python/pyspark/pandas/plot/core.py#L142.

Reply:
Modified to accept NumericType data only, thank you!

Reply:
let's also adjust the PR description